web/src/layout/map.js

```diff
@@ -146,16 +147,37 @@ export default () => {
   );
 
   const handleViewportChange = useMemo(
-    () => ({ latitude, longitude, zoom }) => {
+    () => ({
+      width,
+      height,
+      latitude,
+      longitude,
+      zoom,
+    }) => {
       dispatchApplication('isMovingMap', true);
-      dispatchApplication('mapViewport', { latitude, longitude, zoom });
+      dispatchApplication('mapViewport', {
+        width,
+        height,
+        latitude,
+        longitude,
+        zoom,
+      });
       // TODO: Try tying this to internal map state
       // somehow to remove the need for debouncing.
       debouncedReleaseMoving();
     },
     [],
   );
 
+  // Update the map viewport whenever the window width or height changes.
+  // TODO: This is a hack that should ideally not be needed, but without it
+  // the exchange arrows don't seem to update well under resizing the width.
+  // See https://github.com/tmrowco/electricitymap-contrib/issues/2573#issuecomment-658242753
+  const { width, height } = useWindowSize();
+  useEffect(() => {
+    handleViewportChange({ ...viewport, width, height });
+  }, [width, height]);
+
```

Comment:
Do you know if this affects performance? I know we have had a lot of issues with the exchange arrows and the performance in the past.

Even still this is probably still the best solution.

Just curious, in the bug found by q--, it seemed to only occur when dragging horizontally, that is when the width changes and the height stay constant. Do we then need to use this "hack" for both height and width changes, or could we just use it for width changes only?

Reply:
> Do you know if this affects performance?

I tested it quickly and it seems like the answer is not really. Also, I think it's important to note that window resizing doesn't happen that often and the performance in fixed viewport should be completely unaffected by this code.

> Do we then need to use this "hack" for both height and width changes, or could we just use it for width changes only?

You're right, in this particular case I think we can remove the height listener but I decided to keep it as I wasn't entirely sure if not tracking the width is a bug or rather tracking the height is an accidental _feature_ so I added both here to be on the safe side regarding the future changes :slightly_smiling_face:

Reply:
Good points, thanks :)